src/Authenticator/ChangePasswordHandler.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace SilverStripe\MFA\Authenticator;
+
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\MFA\RequestHandler\BaseHandlerTrait;
+use SilverStripe\MFA\RequestHandler\LoginHandlerTrait;
+use SilverStripe\MFA\Service\MethodRegistry;
+use SilverStripe\MFA\Service\SchemaGenerator;
+use SilverStripe\MFA\Store\StoreInterface;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\MemberAuthenticator\ChangePasswordHandler as BaseChangePasswordHandler;
+
+class ChangePasswordHandler extends BaseChangePasswordHandler
+{
+    use BaseHandlerTrait;
+    use LoginHandlerTrait;
+
+    private static $url_handlers = [
+        'GET mfa/schema' => 'getSchema', // Provides details about existing registered methods, etc.
+        'GET mfa/login/$Method' => 'startMFACheck', // Initiates login process for $Method
+        'POST mfa/login/$Method' => 'verifyMFACheck', // Verifies login via $Method
+        'GET mfa' => 'mfa', // Renders the MFA Login Page to init the app
+    ];
+
+    private static $allowed_actions = [
+        'changepassword',
+        'mfa',
+        'getSchema',
+        'startMFACheck',
+        'verifyMFACheck',
+    ];
+
+    protected function jsonResponse(array $response, $code = 200)
+    {
+        return HTTPResponse::create(json_encode($response))
+            ->addHeader('Content-Type', 'application/json')
+            ->setStatusCode($code);
+    }
+
+    public function getSchema()
+    {
+        try {
+            $member = $this->getStore()->getMember();
+            $schema = SchemaGenerator::create()->getSchema($member);
+            return $this->jsonResponse(
+                array_merge($schema, [
+                    'endpoints' => [
+                        'login' => $this->Link('mfa/login/{urlSegment}'),
+                        'complete' => $this->Link(),
+                    ],
+                    'shouldRedirect' => false,
+                ])
+            );
+        } catch (Throwable $exception) {
+            // If we don't have a valid member we shouldn't be here...
```

Comment:
We have a logger now for this (see the readme) - perhaps log the exception you catch